lib/Sema/TypeCheckType.cpp

```diff
@@ -305,9 +305,11 @@ Type TypeChecker::resolveTypeInContext(
       continue;
 
     // Search the type of this context and its supertypes.
-    for (auto fromType = resolver->resolveTypeOfContext(parentDC);
-         fromType;
-         fromType = getSuperClassOf(fromType)) {
+    auto fromType = resolver->resolveTypeOfContext(parentDC);
+    for (;;) {
```

Comment:
@dabrahams Good question! :-)

I wanted to keep the code 100 % functionally equivalent with the exception of the added `assert`-line. Specifically I wanted to avoid calling `getSuperClassOf(…)` twice (once in the for-loop afterthought and once in the `assert`).

Can you think of a way to write it that avoids calling `getSuperClassOf(…)` twice while still keeping the for-loop intact? Or perhaps calling `getSuperClassOf(…)` twice is acceptable? :-)